test/e2e/network/network_policy.go

```diff
@@ -352,12 +486,552 @@ var _ = SIGDescribe("NetworkPolicy", func() {
+					// Apply this policy to allow all pods
+					PodSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{},
+					},
+					PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+					Ingress:     []networkingv1.NetworkPolicyIngressRule{{}},
+				},
+			}
+
+			policyAllowAll, err = f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(policyAllowAll)
+			Expect(err).NotTo(HaveOccurred())
+			defer cleanupNetworkPolicy(f, policyAllowAll)
+
+			// client-a should be able to contact server as the policy is to allow all
+			By("Creating client-a which should be able to contact the server.", func() {
+				testCanConnect(f, f.Namespace, "client-a", service, 80)
+			})
+		})
+
+		It("should enforce policies and test impact of deletion of policy on pod traffic based on PodSelector [Feature:NetworkPolicy]", func() {
+			By("Creating a network policy for the server which denies all traffic.")
+			policyDenyAll := &networkingv1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: f.Namespace.Name,
+					Name:      "deny-all",
+				},
+				Spec: networkingv1.NetworkPolicySpec{
+					// Deny all traffic
+					PodSelector: metav1.LabelSelector{},
+					PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+					Ingress:     []networkingv1.NetworkPolicyIngressRule{},
+				},
+			}
+
+			policyDenyAll, err := f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(policyDenyAll)
+			Expect(err).NotTo(HaveOccurred())
+			defer cleanupNetworkPolicy(f, policyDenyAll)
+
+			By("Creating client-a which should not be able to contact the server.", func() {
+				testCannotConnect(f, f.Namespace, "client-a", service, 80)
+			})
+
+			By("Creating a network policy for the server which allows traffic only from client-a.")
+			policyAllowClientA := &networkingv1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: f.Namespace.Name,
+					Name:      "allow-from-client-a-pod-selector",
+				},
+				Spec: networkingv1.NetworkPolicySpec{
+					// Apply this policy to the Server
+					PodSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"pod-base-name": podServerLabelSelector,
+						},
+					},
+					PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+					// Allow traffic from all pods
+					Ingress: []networkingv1.NetworkPolicyIngressRule{{
+						From: []networkingv1.NetworkPolicyPeer{{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"pod-base-name": "client-a",
+								},
+							},
+						}},
+					}},
+				},
+			}
+
+			policyAllowClientA, err = f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(policyAllowClientA)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating client-a which should be able to contact the server.", func() {
+				testCanConnect(f, f.Namespace, "client-a", service, 80)
+			})
+
+			cleanupNetworkPolicy(f, policyAllowClientA)
```

Comment:
need a `By("deleting the network policy allowing traffic from client-a")` here

Reply:
Acknowledged and added the By().